### DISCUSSION TITLE
Error: Error: spawn Unknown system error -86 (on Apple Silicon)

### DISCUSSION THREAD

**Discussion by EntityIT** [microsoft/vscode-cpptools#7298] [Q&A] [answered]
I have recently installed Visual Studio Code (version 1.55.0) and C/C++ extension (version 1.3.0-insiders3) on my MacBook Air (M1, 2020). However, after the installation of C/C++ extension (version 1.3.0-insiders3), I saw 2 notifications, which are "Couldn't start client cpptools" and "Unable to start the C/C++ language server. IntelliSense features will be disabled. Error: Error: spawn Unknown system error -86". I don't know what caused this and cannot find a way to fix. Please help!
![image](https://user-images.githubusercontent.com/40011478/113727119-09418000-971f-11eb-8c70-64e61d80feec.png)
![image](https://user-images.githubusercontent.com/40011478/113728506-5e31c600-9720-11eb-8a49-8355b9d91f60.png)

**Comment by michelleangela** [comment, marked as answer]
Can you verify if this is also an issue with release version 1.2.2 on your machine? We have another report of this at https://github.com/microsoft/vscode-cpptools/issues/7286. We have not identified the root cause, but it could be the same.

**Reply by EntityIT** [reply]
I verify that this is also an issue with release version 1.2.2 on my machine.

**Reply by EntityIT** [reply]
I found a solution. Mine works fine after installing Rosetta.

**Reply by michelleangela** [reply]
Does that mean the Rosetta emulator for x64 programs is not installed by default on Apple Silicon or it got disabled if the extension was working before?

Just a note that we're soon to release a 1.3.0-insiders4 version of the extension that will have ARM64 Mac binaries that can run natively on Apple Silicon. By using the ARM64 binaries, it wouldn't need to use Rosetta.

**Reply by EntityIT** [reply]
Yes, the Rosetta emulator for x64 programs is not installed by default on Apple Silicon. It is suggested to be installed when I try to open an Intel-based app. And I'm interested to wait for the 1.3.0-insiders4 version.

**Comment by michelleangela** [comment]
The native language service binaries for ARM64 Mac is now available on our 1.3.0-insiders4 version as VSIX package **cpptools-osx-arm64.vsix** from https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0-insiders4
